reana_server/reana_admin/check_workflows.py

```diff
@@ -146,73 +163,141 @@ def check_workflows(
+    ]
+}
+
+
+def check_interactive_sessions() -> Tuple[
+    List[CheckResult], List[CheckResult], List[CheckResult], int
+]:
+    """Check if selected sessions are in sync with database according to predefined rules.
+
+    Because interactive sessions are deleted from database when they are closed,
+    the function queries all of them at once.
+    The number of interactive sessions is usually low (<100) so applying time range doesn't make sense.
+    In addition, session pods can still exist even if session is not present in the database.
+    So, it is easier to query all sessions to determine if some pods are hanging.
+    """
+    statuses_to_check = interactive_sessions_validation_map.keys()
+    filtered_sessions = Session.query(InteractiveSession).filter(
+        InteractiveSession.status.in_(statuses_to_check)
+    )
+
+    total_sessions = filtered_sessions.count()
+
+    click.secho(
+        f"Found {total_sessions} interactive session(s) in the database.",
+    )
+
+    try:
+        pods = _get_all_pods()
+    except ApiException as error:
+        raise InfoCollectionError(f"Couldn't fetch list of pods: {error}")
+
+    in_sync_sessions: List[CheckResult] = []
+    out_of_sync_sessions: List[CheckResult] = []
+    pods_without_session: List[CheckResult] = []
+
+    # check if some session pods don't have corresponding session entry in the database
+    session_pods = [pod for pod in pods if "run-session" in pod.metadata.name]
+    for pod in session_pods:
+        pod_doesnt_have_session = not any(
+            [session.name in pod.metadata.name for session in filtered_sessions]
         )
-        _display(in_sync_workflows)
+        if pod_doesnt_have_session:
+            check_source = CheckSource(
+                id=pod.metadata.name, name="Session pod", user=None, status=None
+            )
+            failed_checks = [CheckFailed("Pod doesn't have a session in the database.")]
+            pods_without_session.append(CheckResult(check_source, failed_checks))
+
+    # check if sessions in the database pass checks
+    for session in filtered_sessions:
+        filtered_pods = [
+            pod for pod in session_pods if str(session.name) in pod.metadata.name
+        ]
+
+        checks = interactive_sessions_validation_map[session.status]
+        failed_checks: List[CheckFailed] = []
+        session_is_in_sync = True
 
-    if out_of_sync_workflows:
-        click.secho(
-            f"\nOut-of-sync workflows ({len(out_of_sync_workflows)} out of {filtered_workflows.count()})\n",
-            fg="red",
+        for check in checks:
+            try:
+                check(session, filtered_pods, [])
+            except CheckFailed as error:
+                failed_checks.append(error)
+                session_is_in_sync = False
+                break
+
+        check_source = CheckSource(
+            id=str(session.id_),
+            name=session.name,
+            user=str(session.owner_id),
```

Comment:
```suggestion
            user=str(session.workflow[0].owner.email),
```
WDYT? This is not the best, but it should work fine as a session is always attached to a workflow.